tests/e2e/demo-components.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { readdirSync } from 'fs';
+import { expect, test } from '@playwright/test';
+
+const runComponentTests = (testConfig) => {
+  const { packageDir, componentsDir, host, srcDir = '', hiddenComponents = [], type } = testConfig;
+  const name = type
+    .split('-')
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ');
+
+  test.describe(`Demo ${name} Components`, () => {
+    const componentDirs = readdirSync(`${packageDir}${srcDir}${componentsDir}`, { withFileTypes: true })
+      .filter((item) => item.isDirectory())
+      .filter((item) => !hiddenComponents.includes(item.name))
+      .map((item) => item.name.toLowerCase());
+
+    for (const component of componentDirs) {
+      test(`test demo ${name} component ${component}`, async ({ page }) => {
+        await page.goto(`${host}${componentsDir}/${component}/${type === 'web-twig' ? '?HIDE_TOOLBAR' : ''}`);
+        // wait for fonts to load
+        await page.evaluate(() => document.fonts.ready);
+        // wait for transitions to finish
+        await page.waitForLoadState();
+        // disable animations to avoid flaky screenshots
+        await page.addStyleTag({ content: '*, *::before, *::after { animation-iteration-count: 1 !important }' });
+        await expect(page).toHaveScreenshot(`${type}-${component}.png`, {
+          animations: 'disabled',
+          fullPage: true,
+        });
+      });
+    }
+  });
+};
+
+const testConfigs = [
+  {
+    componentsDir: '/src/scss/components',
+    host: 'http://192.168.64.1:5175',
```

Comment:
These hosts have to be configured from one config. It should be solved in #1186

Reply:
This has been solved.